internal/services/services.go

```diff
@@ -46,6 +61,81 @@ func (dm *defaultMetricsService) AllCounters(f func(string, int64)) {
 	dm.storage.AllCounters(f)
 }
 
+func (dm *defaultMetricsService) saveCounterStruct(m *model.Metrics) (*model.Metrics, error) {
+	result := &model.Metrics{ID: m.ID, MType: m.MType}
+	if strings.TrimSpace(m.ID) == "" {
+		return nil, ErrIDRequired
+	}
+	if m.Delta == nil {
+		return nil, ErrDeltaRequired
+	}
+	newDelta := dm.SaveCounter(m.ID, *m.Delta)
+	result.Delta = &newDelta
+	return result, nil
+}
+
+func (dm *defaultMetricsService) saveGaugeStruct(m *model.Metrics) (*model.Metrics, error) {
```

Comment:
```suggestion
func (dm *defaultMetricsService) saveGauge(m *model.Metrics) (*model.Metrics, error) {
```
Struct не несет нагрузки